_posts/2020-06-05-mirror-maker-2-eventhub.md

```diff
@@ -0,0 +1,290 @@
+
+```
+bootstrap.servers=<eventhubs-namespace>.servicebus.windows.net:9093
+security.protocol=SASL_SSL
+sasl.mechanism=PLAIN
+sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="$ConnectionString" password="Endpoint=sb://<eventhubs-namespace>.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=<access-key>";
+```
+
+Save the above properties into a `kafka_eventhub.properties` file and start the consumer as following.
+
+```shell
+bin/kafka-console-consumer.sh --bootstrap-server <eventhubs-namespace>.servicebus.windows.net:9093 --topic testeh --consumer.config kafka_eventhub.properties
+```
+
+To try the demo, the only thing left to do is to send some messages.
+To do so, just use the `kafka-console-producer` command line tool provided with Apache Kafka.
+Start a new pod in the Kubernetes cluster for hosting the producer and type a couple of messages as follows.
+
+```shell
+kubectl -n kafka run kafka-producer -ti --image=strimzi/kafka:0.18.0-kafka-2.5.0 --rm=true --restart=Never -- bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap:9092 --topic testeh
+If you don't see a command prompt, try pressing enter.
+>"Hello from Strimzi Mirror Maker 2"
+>"Here another mirrored message"
+>
+```
+
+On the Kafka console consumer application, the messages will be logged like this:
+
+```shell
+"Hello from Strimzi Mirror Maker 2"
+"Here another mirrored message"
+```
+
+### Let's go "active-active"
+
+There are some scenarios where it's needed having a bidirectional mirroring in order to have producer and consumer on both sides, consuming data on the "same" topic.
+
+![Overall active-active architecture](/assets/images/posts/2020-06-05-mirror-maker-2-eventhub-active-active.png)
+
+In the previous unidirectional use case, the producer writes to the `testeh` topic on the local Kafka cluster and it's mirrored to the `my-cluster.testeh` on Azure Event Hub from where the consumer gets messages.
+If we want to produce messages to a `testeh` event hub on Azure Event Hub namespace and mirroring this one to the local Kafka cluster, we have to setup another `KafkaMirrorMaker2` instance where source and target are switched.
+In this case the destination topic on the local Kafka cluster will be `eventhub.testeh`.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: KafkaMirrorMaker2
+metadata:
+  name: my-mm2-cluster-eh
+spec:
+  version: 2.5.0
+  replicas: 1
+  connectCluster: "my-cluster"
+  clusters:
+  - alias: "my-cluster"
+    bootstrapServers: my-cluster-kafka-bootstrap:9092
+  - alias: "eventhub"
+    bootstrapServers: <eventhubs-namespace>.servicebus.windows.net:9093
+    config:
+      config.storage.replication.factor: 1
+      offset.storage.replication.factor: 1
+      status.storage.replication.factor: 1
+      producer.connections.max.idle.ms: 180000
+      producer.metadata.max.age.ms: 180000
```

Comment:
Since this is now consumer, should these options change?

Reply:
Yes, another copy/paste mistake :P